File: API/Entities/Class.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities
{
    [Table("Class")]
    public class Class
    {
        public int Id { get; set; }
        [Required]
        public int ClassNumber { get; set; }
        [Required]
        public string ClassName { get; set; }
        public string Description { get; set; }
        [Required]
        public int ClassWaitingPeriod { get; set; }
        [Required]
        public double PersonalHealthMaximum { get; set; }
        public bool IsHsaClass { get; set; }
        public Company Company { get; set; }
        public int CompanyId { get; set; }

    }
}

File: client/src/app/_models/class.ts
import { HsaClassDetails } from "./hsaclassdetails";

export interface Class {
    id: number;
    classNumber: number;
    className: string;
    description: string;
    classWaitingPeriod: number;
    personalHealthMaximum: number;
    isHsaClass: boolean;
    companyId: number;
    hsaClassDetails: HsaClassDetails[];
}